api/func-e.go

```diff
@@ -77,20 +104,39 @@ type runOpts struct {
 // This blocks until the context is done or the process exits. The error might be
 // context.Canceled if the context is done or an error from the process.
 func Run(ctx context.Context, args []string, options ...RunOption) error {
-	ro := &runOpts{out: os.Stdout}
+	// TODO: we need a real API and it being an interface in this package, initialized in the root
+	// directory like wazero does. That this stitches the impl makes it not an API package and causes
+	// package import complexity we need to remove.
+	o, err := initOpts(ctx, options...)
+	if err != nil {
+		return err
+	}
+	return api.Run(ctx, o, args)
+}
+
+// initOpts is a placeholder to adapt E2E tests until we have a real API for func-e.
+func initOpts(ctx context.Context, options ...RunOption) (*globals.GlobalOpts, error) {
+	ro := &runOpts{
+		out:      os.Stdout,
+		envoyOut: os.Stdout,
+		envoyErr: os.Stderr,
+	}
 	for _, option := range options {
 		option(ro)
 	}
 
-	o := globals.GlobalOpts{
-		HomeDir:          ro.homeDir,
-		EnvoyVersion:     version.PatchVersion(ro.envoyVersion),
-		EnvoyVersionsURL: ro.envoyVersion,
-		Out:              ro.out,
+	o := &globals.GlobalOpts{
+		EnvoyVersion: version.PatchVersion(ro.envoyVersion),
+		Out:          ro.out,
+		RunOpts: globals.RunOpts{
+			EnvoyPath: ro.envoyPath,
+			EnvoyOut:  ro.envoyOut,
+			EnvoyErr:  ro.envoyErr,
+		},
+	}
+	if err := api.InitializeGlobalOpts(o, ro.envoyVersionsURL, ro.homeDir, ""); err != nil {
+		return nil, err
 	}
 
-	funcECmd := cmd.NewApp(&o)
```

Comment:
before, this routed library logic through urfave which was creepy and made it near impossible to understand